### PR TITLE
Implement YADS concat

### PR DESCRIPTION
**Pull request by toteto-h4**
This PR implements `concat` method for `MArray`. The behavior and signature can be expected to be same as `Array.concat`.

### CODE REVIEW

File: src/tree-utils.ts [hunk 1]
```diff
@@ -267,6 +267,9 @@ export function* iterate<T>(
   index: number = 0,
   count: number = root.getField(Size),
 ) {
+  if (root.getField(Size) === 0) {
+    return;
+  }
```
Comment (toteto-h4) [inline src/tree-utils.ts:272]: @spion-h4 this is the part that fixes the creation and iteration of empty yads tree.

File: src/tree-utils.ts [hunk 2]
```diff
@@ -267,6 +267,9 @@ export function* iterate<T>(
...
   if (index < 0) {
```
Comment (spion-h4) [inline src/tree-utils.ts:273]: note: we should also remove this behavior in the future

**Review by spion-h4 — APPROVED** [review]

File: tests/structure.test.ts [hunk 3]
```diff
@@ -16,8 +16,8 @@ describe('Internal tree structure', () => {
   });
 
   it('Creates a tree', () => {
-    const data: number[] = [0];
-    for (let i = 1; i < 11; ++i) {
```
Comment (korun-h4) [inline tests/structure.test.ts:20]: Is there any impact of this change function wise, or just style preference?

Comment (spion-h4) [reply tests/structure.test.ts:20]: it starts from 0-length arrays now, which means we support empty MVectors

**Pull request merged**